lib/workers/bin/provider_worker

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../../config/environment", __dir__)
+
+ems_id = ENV["EMS_ID"]
+
+ems = ExtManagementSystem.find(ems_id)
+
+connection_config = {
+  :ems_id          => ems_id,
+  :ems_name        => ems.name,
+  :endpoints       => ems.endpoints,
+  :authentications => ems.authentications,
+  :messaging_opts  => MiqQueue.send(:messaging_client_options)
+}
+
+stdin_rd, stdin_wr = IO.pipe
+stdin_wr.write(connection_config.to_json)
+stdin_wr.close
+
+$stdin.reopen(stdin_rd)
+
+Kernel.exec(ARGV.join(" "))
```

Comment:
Should this use shelljoin? https://ruby-doc.org/stdlib-2.5.1/libdoc/shellwords/rdoc/Shellwords.html#method-c-shelljoin

Reply:
:+1: nice, yes it should done